src/dxtbx/format/FormatNXmxDLS.py

```diff
@@ -13,36 +12,63 @@
 from dxtbx.format.FormatNXmx import FormatNXmx
 
 
-def get_bit_depth_from_meta(meta_file_name):
+def get_bit_depth_from_meta(meta_file_name: Path) -> int:
     with h5py.File(meta_file_name) as f:
-        return int(f["/_dectris/bit_depth_image"][()])
+        with contextlib.suppress(KeyError):
+            return int(f["/_dectris/bit_depth_image"][()])
 
+        # This might be a very old meta file, that only had 'datatype'?
+        if "datatype" in f:
+            unique_vals = np.unique(f["datatype"])
```

Comment:
Sensible, but not sure it is necessary i.e. just return the biggest?

Reply:
I'm not confident I know the implications of that for handling multi-data-type datasets, and we don't have any test data for it so I don't know if doing that would make things work. I'd rather keep this working on the simple case and address any other failures later.